src/protocols/rdp/channels/cliprdr.c

```diff
@@ -614,6 +619,11 @@ int guac_rdp_clipboard_end_handler(guac_user* user, guac_stream* stream) {
     if (clipboard == NULL)
         return 0;
 
+    /* Don't need to send it to the remote server, if the local clipboard is
+     * empty */
+    if (clipboard->clipboard->length == 0)
+        return 0;
```

Comment:
No, it can never happen. I think that we don't need to update the remote clipboard when there is no data in the local clipboard.

Reply:
It looks like it can:

https://www.windowscentral.com/how-clear-clipboard-data-shortcut-windows-10

Testing the suggestion at the link above (running `echo off | clip` from within the Windows command line), that does indeed clear the clipboard, and Guacamole currently handles that correctly.

Reply:
Is this related to this pull request?

Reply:
Yes. In response to Nick asking noting that this change would remove the ability for a user to clear the remote clipboard, you said:

> No, it can never happen. I think that we don't need to update the remote clipboard when there is no data in the local clipboard.

On the contrary, from the above link:

* It can happen.
* It is desirable to be able to do so for the sake of privacy.

If the local clipboard is intentionally cleared, the remote clipboard needs to be cleared, as well.